octodns_ns1/__init__.py

```diff
@@ -1744,6 +1744,24 @@ def _extra_changes(self, desired, changes, **kwargs):
 
         return extra
 
+    def _force_root_ns_update(self, changes):
+        '''
+        Changes any 'Create' changetype for a root NS record to an 'Update'
+        changetype. Used on new zone creation, since NS1 will automatically create root NS records (see https://ns1.com/api?docId=2184).
+        This means our desired NS records must be applied as an Update, rather than a Create.
+        '''
+        for i, change in enumerate(changes):
+            if (
+                change.record.name == ''
+                and change.record._type == 'NS'
+                and isinstance(change, Create)
+            ):
+                self.log.info(
+                    '_force_root_ns_update: found root NS record creation, changing to update'
+                )
+                changes[i] = Update(None, change.record)
+        return
```

Comment:
```suggestion
        return changes
```

Minor tweak to just make it inherently obvious that _force_root_ns_update is going to be modifying changes.